crates/turbopath/src/absolute_system_path_buf.rs

```diff
@@ -0,0 +1,175 @@
+use std::{
+    borrow::Cow,
+    ffi::OsStr,
+    fmt,
+    path::{Components, Path, PathBuf},
+};
+
+use serde::Serialize;
+
+use crate::{AnchoredSystemPathBuf, IntoSystem, PathValidationError, RelativeSystemPathBuf};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Serialize)]
+pub struct AbsoluteSystemPathBuf(PathBuf);
+
+impl AbsoluteSystemPathBuf {
+    /// Create a new AbsoluteSystemPathBuf from `unchecked_path`.
+    /// Confirms that `unchecked_path` is absolute and converts it to a system
+    /// path.
+    ///
+    /// # Arguments
+    ///
+    /// * `unchecked_path`: The path to be validated and converted to an
+    ///   `AbsoluteSystemPathBuf`.
+    ///
+    /// returns: Result<AbsoluteSystemPathBuf, PathValidationError>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::{Path, PathBuf};
+    /// use turbopath::AbsoluteSystemPathBuf;
+    /// let path = PathBuf::from("/Users/user");
+    /// let absolute_path = AbsoluteSystemPathBuf::new(path).unwrap();
+    /// #[cfg(windows)]
+    /// assert_eq!(absolute_path.as_path(), Path::new("\\Users\\user"));
```

Comment:
I think this will fail on windows since a drive is required for a path to be absolute: https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.is_absolute